cloud/services/compute/network.go

```diff
@@ -64,6 +65,14 @@ func (s *Service) ReconcileNetwork() error {
 	s.scope.GCPCluster.Spec.Network.Name = pointer.StringPtr(network.Name)
 	s.scope.GCPCluster.Spec.Network.AutoCreateSubnetworks = pointer.BoolPtr(network.AutoCreateSubnetworks)
 	s.scope.GCPCluster.Status.Network.SelfLink = pointer.StringPtr(network.SelfLink)
+
+	// Create subnetworks if the network is created in custom mode
+	if !network.AutoCreateSubnetworks {
+		if err := s.createSubnetworks(); err != nil {
```

Comment:
what if AutoCreateSubnetworks==false, and no subnet provided in gcpcluster.spec?

Reply:
In that case the subnets wont be created and the flow will continue. But instances need subnet either provided or the first available subnet in the region is picked. But here as subnets are not created instance creation would fail

Reply:
While creating and deleting subnetworks I should also check if AutoCreateSubnetworks exist in the ForceSendFields so as to differentiate between legacy and custom network. However there is no impact right now as the subnet list would be empty and no operation would be performed in legacy mode. I'll make the change anyways.